File: 1217-relative-sort-array/relative-sort-array.go
func relativeSortArray(arr1 []int, arr2 []int) []int {
    count := make(map[int]int)
	for _, num := range arr1 {
		count[num]++
	}

	res := make([]int, 0)
	for _, num := range arr2 {
		for i := 0; i < count[num]; i++ {
			res = append(res, num)
		}
		delete(count, num)
	}

	remaining := make([]int, 0)
	for num := range count {
		remaining = append(remaining, num)
	}
	sort.Ints(remaining)

	for _, num := range remaining {
		for i := 0; i < count[num]; i++ {
			res = append(res, num)
		}
	}

	return res
}

File: 1217-relative-sort-array/relative-sort-array.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        count = [0] * (max(arr1) + 1)
        for num in arr1:
            count[num] += 1
        
        res = []
        for num in arr2:
            res.extend([num] * count[num])
            count[num] = 0
        
        for i, num in enumerate(count):
            if num > 0:
                res.extend([i] * num)
        
        return res